### DISCUSSION TITLE
Issuer.discover() vs Issuer.webfinger(), naming, & "OpenID Provider Issuer Discovery"

### DISCUSSION THREAD

**Discussion by alxndrsn** [panva/openid-client#614] [Q&A] [answered]
Thanks for a great library.

The docs for `Issuer.webfinger()` seem very clear:

> Performs [OpenID Provider Issuer Discovery](https://openid.net/specs/openid-connect-discovery-1_0.html#IssuerDiscovery) based on End-User input.

And `Issuer.discover()`:

> Loads OpenID Connect 1.0 and/or OAuth 2.0 Authorization Server Metadata documents.

So `webfinger()` discovers, and `discover()` fetches metadata.  Have I understood correctly?  Is there any appetite for renaming the `discover()` function?  I'm not sure what might be more accurate, maybe `Issuer.fromUrl()`?

**Comment by panva** [comment, marked as answer]
Hi @alxndrsn 

> Is there any appetite for renaming the discover() function?

There isn't. If anything the webfinger method is going to be removed in the next major on the account of trimming the API surface of things that are barely ever used.

**Comment by alxndrsn** [comment]
Thanks @panva :+1:
